javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -0,0 +1,1131 @@
+      override protected val inner: Collection[E] = c
+
+      override def equals(obj: Any): Boolean =
+        this eq obj.asInstanceOf[AnyRef]
+
+      override def hashCode(): Int =
+        System.identityHashCode(this)
+    }
+    c match {
+      case _: Serializable => new BasicCheckedCollection with Serializable
+      case _               => new BasicCheckedCollection
+    }
+  }
+
+  def checkedSet[E](s: Set[E], typ: Class[E]): Set[E] = {
+    class BasicCheckedSet extends CheckedSet[E, Set[E]] {
+      override protected val elemClazz: Class[E] = typ
+      override protected val inner: Set[E] = s
+    }
+    s match {
+      case _: Serializable => new BasicCheckedSet with Serializable
+      case _               => new BasicCheckedSet
+    }
+  }
+
+  def checkedSortedSet[E](s: SortedSet[E], typ: Class[E]): SortedSet[E] = {
+    class BasicCheckedSortedSet extends CheckedSortedSet[E] {
+      override protected val elemClazz: Class[E] = typ
+      override protected val inner: SortedSet[E] = s
+    }
+    s match {
+      case _: Serializable => new BasicCheckedSortedSet with Serializable
+      case _               => new BasicCheckedSortedSet
+    }
+  }
+
+  def checkedList[E](list: List[E], typ: Class[E]): List[E] = {
+    class BasicCheckedList extends CheckedList[E] {
+      protected val elemClazz: Class[E] = typ
+      protected val inner: List[E] = list
+    }
+    list match {
+      case _: Serializable with RandomAccess =>
+        new BasicCheckedList with Serializable with RandomAccess
+      case _: Serializable => new BasicCheckedList with Serializable
+      case _: RandomAccess => new BasicCheckedList with RandomAccess
+      case _               => new BasicCheckedList
+    }
+  }
+
+  def checkedMap[K, V](m: Map[K, V], keyType: Class[K], valueType: Class[V]): Map[K, V] = {
+    class BasicCheckedMap extends CheckedMap[K, V, Map[K, V]] {
+      protected def keyClazz: Class[K] = keyType
+      protected def valueClazz: Class[V] = valueType
+      protected val inner: Map[K, V] = m
+    }
+    m match {
+      case _: Serializable => new BasicCheckedMap with Serializable
+      case _               => new BasicCheckedMap
+    }
+  }
+
+  def checkedSortedMap[K, V](m: SortedMap[K, V], keyType: Class[K], valueType: Class[V]): SortedMap[K, V] = {
+    class BasicCheckedSortedMap extends CheckedSortedMap[K, V] {
+      protected def keyClazz: Class[K] = keyType
+      protected def valueClazz: Class[V] = valueType
+      protected val inner: SortedMap[K, V] = m
+    }
+    m match {
+      case _: Serializable => new BasicCheckedSortedMap with Serializable
+      case _               => new BasicCheckedSortedMap
+    }
+  }
+
+  def emptyIterator[T](): Iterator[T] = {
+    new Iterator[T] {
+      override def hasNext: Boolean = false
+
+      override def next(): T =
+        throw new NoSuchElementException
+
+      override def remove(): Unit =
+        throw new IllegalStateException
+    }
```

Comment:
Consider declaring a singleton instance `EmptyIterator` and cast it to `Iterator[T]` in this method.